mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ProxyConfigurationResourceDefinition.java

```diff
@@ -343,14 +346,40 @@ public ManagementResourceRegistration register(ManagementResourceRegistration pa
         ManagementResourceRegistration registration = parent.registerSubModel(this);
 
         ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver())
-                .addAttributes(EnumSet.complementOf(EnumSet.of(Attribute.SSL_CONTEXT)))
-                .addExtraParameters(Attribute.SSL_CONTEXT)
-                .addCapabilities(Capability.class);
+                .addAttributes(EnumSet.complementOf(EnumSet.of(Attribute.SIMPLE_LOAD_PROVIDER, Attribute.SSL_CONTEXT)))
+                .addExtraParameters(Attribute.SIMPLE_LOAD_PROVIDER, Attribute.SSL_CONTEXT)
+                .addCapabilities(Capability.class)
+                ;
+        registration.registerReadWriteAttribute(Attribute.SIMPLE_LOAD_PROVIDER.getDefinition(), new OperationStepHandler() {
+                    @Override
+                    public void execute(OperationContext context, ModelNode operation) {
+                        Resource resource = context.readResource(PathAddress.EMPTY_ADDRESS, true);
+                        if (resource.hasChild(SimpleLoadProviderResourceDefinition.PATH)) {
+                            ModelNode simple = resource.getChild(SimpleLoadProviderResourceDefinition.PATH).getModel();
+                            context.getResult().set(simple.get(SimpleLoadProviderResourceDefinition.Attribute.FACTOR.getName()));
+                        } else {
+                            context.getResult();
+                        }
+                    }
+                }
+                , new ReloadRequiredWriteAttributeHandler() {
+                    @Override
+                    protected void validateUpdatedModel(OperationContext context, Resource resource) {
+                        context.addStep(new OperationStepHandler() {
+                            @Override
+                            public void execute(OperationContext ctx, ModelNode op) {
+                                if (!resource.hasChild(SimpleLoadProviderResourceDefinition.PATH)) {
```

Comment:
This seems unnecessarily cumbersome/error-prone.
I suggest making SimpleLoadProviderResourceDefintion the default child of this type via ResourceDescriptor.addRequiredSingletonChildren(SimpleLoadProviderResourceDefinition.PATH)).
That way it will already exist when this write-attribute handler executes, and will auto-remove if the dynamic load provider is added.
This means you can delegate to the existing write attribute handler for the child resource's attribute.

Reply:
Done using translation mechanism.